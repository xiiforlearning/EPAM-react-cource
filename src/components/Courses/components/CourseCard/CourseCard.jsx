import React from 'react';
import { useNavigate } from 'react-router-dom';
import { useSelector } from 'react-redux';

import { Button } from '../../../../common/Button/Button';
import { BUTTON_TEXT_SHOW_COURSE } from '../../../../constants';
import {
	dateConvert,
	getAutors,
	timeConvert,
} from '../../../../helpers/getCourseInfo';

import styles from './courseCard.module.scss';
import { removeCourseAction } from '../../../../store/course/thunk';
import store from '../../../../store';

const CourseCard = ({
	title,
	description,
	creationDate,
	duration,
	authors,
	allAuthors,
	id,
}) => {
	const navigate = useNavigate();
	const authorsList = getAutors(authors, allAuthors).join(', ');
	const role = useSelector((state) => state.user.role);

	return (
		<div className={styles.wrapper} data-testid='course'>
			<div className={styles.courseDesc}>
				<h2>{title}</h2>
				<p>{description}</p>
			</div>
			<div className={styles.courseInfoWrapper}>
				<div className={styles.courseInfo}>
					<div>
						<span>Autors: </span>
						<p>{authorsList}</p>
					</div>
					<div>
						<span>Duration: </span>
						<p>{duration && timeConvert(duration)}</p>
					</div>
					<div>
						<span>Created: </span>
						<p>{creationDate && dateConvert(creationDate)}</p>
					</div>
				</div>
				<div className={styles.buttonsWrapper}>
					<Button
						text={BUTTON_TEXT_SHOW_COURSE}
						onClick={() => {
							navigate(`/courses/${id}`);
						}}
					/>
					{role === 'admin' && (
						<>
							<Button
								text='delete'
								onClick={() => {
									store.dispatch(removeCourseAction(id));
								}}
							/>
							<Button
								text='update'
								onClick={() => {
									navigate(`/courses/update/${id}`);
								}}
							/>
						</>
					)}
				</div>
			</div>
		</div>
	);
};

export default CourseCard;
