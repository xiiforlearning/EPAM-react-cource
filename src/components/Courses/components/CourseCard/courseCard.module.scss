.wrapper {
	display: flex;
	justify-content: space-between;
	align-items: center;
	border: 1px solid lightgreen;
	width: 100%;
	padding: 20px;
	gap: 50px;
	align-items: flex-start;
}
.courseDesc {
	flex: 3;
	h2 {
			margin-bottom: 10px;
	}
}
.courseInfoWrapper {
	flex: 1.5;
	display: flex;
	flex-direction: column;
	align-items: flex-start;
	gap: 10px;
	
	button {
			margin: 0 auto;
	}
}
.courseInfo {
	width: 100%;

	div {
		display: flex;
		gap: 10px;

		span {
			font-weight: 700;
			flex: 1;
		}

		p {
			flex: 3;
		}
	}
}