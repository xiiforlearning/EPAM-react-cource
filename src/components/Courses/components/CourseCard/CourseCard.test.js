import React from 'react';
import { create } from 'react-test-renderer';
import { mockedAuthorsList, mockedCoursesList } from '../../../../constants';
import CourseCard from './CourseCard';

const mockedUsedNavigate = jest.fn();
const mockedUsedSelector = jest.fn();

jest.mock('react-router-dom', () => ({
	...jest.requireActual('react-router-dom'),
	useNavigate: () => mockedUsedNavigate,
}));

jest.mock('react-redux', () => ({
	...jest.requireActual('react-redux'),
	useSelector: () => mockedUsedSelector,
}));

describe('CourseCard test', () => {
	it('should render CourseCard component properly', () => {
		const component = create(
			<CourseCard
				title={mockedCoursesList[0].title}
				description={mockedCoursesList[0].description}
				creationDate={mockedCoursesList[0].creationDate}
				duration={mockedCoursesList[0].duration}
				authors={mockedCoursesList[0].authors}
				allAuthors={mockedAuthorsList}
				id={mockedCoursesList[0].id}
			/>
		);

		expect(component.toJSON()).toMatchSnapshot();
	});
});
