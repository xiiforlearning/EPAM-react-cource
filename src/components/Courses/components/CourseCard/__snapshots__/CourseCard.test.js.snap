// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseCard test should render CourseCard component properly 1`] = `
<div
  className="wrapper"
  data-testid="course"
>
  <div
    className="courseDesc"
  >
    <h2>
      JavaScript
    </h2>
    <p>
      Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum 
                    has been the industry's standard dummy text ever since the 1500s, when an unknown 
                    printer took a galley of type and scrambled it to make a type specimen book. It has survived 
                    not only five centuries, but also the leap into electronic typesetting, remaining essentially u
                    nchanged.
    </p>
  </div>
  <div
    className="courseInfoWrapper"
  >
    <div
      className="courseInfo"
    >
      <div>
        <span>
          Autors: 
        </span>
        <p>
          Vasiliy Dobkin, Nicolas Kim
        </p>
      </div>
      <div>
        <span>
          Duration: 
        </span>
        <p>
          2 : 40 hours
        </p>
      </div>
      <div>
        <span>
          Created: 
        </span>
        <p>
          08.03.2021
        </p>
      </div>
    </div>
    <div
      className="buttonsWrapper"
    >
      <button
        onClick={[Function]}
      >
        Show Course
      </button>
    </div>
  </div>
</div>
`;
